### DISCUSSION TITLE
`dev-container-features-test-lib` docs?

### DISCUSSION THREAD

**Discussion by snebjorn** [devcontainers/spec#113] [Q&A] [answered]
Where do I find some docs on dev-container-features-test-lib?

The [feature-template](https://github.com/devcontainers/feature-template/tree/main/test) repo have some usage examples. But it's very mysterious 😄 

Fx. what does `version` do?
```
check "version" color | grep 'my favorite color is red'
```

**Comment by AlexanderLanin** [comment, marked as answer]
I've just encountered the same problem and it took me a while to even find the script, so let's ease that first:
https://github.com/devcontainers/cli/blob/1910ca41015c627b884ddd69ebc52d1e8cdd8cf0/src/spec-node/featuresCLI/utils.ts#L59

An attempt to reverse engineer the commands:
* `check <LABEL> <cmd>`
  executes cmd and prints success/failed depending on exit code of cmd.
  Note: Do not use quotes for cmd as there is no such executable as e.g. "python3 --version".
            Correct: `check "python is available" python3 --version`
* `checkMultiple <LABEL> <MINIMUMPASSED> <cmd1> [cmd2] [cmd3] ...`
  runs all passend commands and prints success/failed depending on passed commands (exit code 0) vs MINIMUMPASSED
* `reportResults`
  prints results of check and checkMultiple

LABEL is basically the test case name

**Reply by joshspicer** [reply]
This is correct, the first argument is the test name/label, and anything after is the argument to the "check" helper.  If the command exits with a 0 exit code, the check succeeds.

Although not officially part of the spec, I will look to add docs on this in the CLI's readme - thanks for raising :)

**Reply by joshspicer** [reply]
Following up here: https://github.com/devcontainers/cli/pull/219
